### PR TITLE
Only update the math when the user is finished

### PR DESCRIPTION
**Pull request by jeffeb3**
This changes the 2 math fields to have two elements in the transform options. One is the "Input" which react uses to own all the state, and update the field as the user is typing, and the other is the one we are used to, but it only gets updated when there is a "blur" (I had to look this up, it is the opposite of focus, when the user leaves the field) or pressing 'Enter'.

This seems to work pretty well. It is tricky to not break the rest of the InputOption boxes when working on this. I am not completely satisfied with the duplication, but I tried to make it simpler, and it just wasn't working as well.

### CODE REVIEW

**Comment by jeffeb3** [comment]
I like that option too, but we want the other inputs to be instantaneous, and we quickly lose the knowledge of what input changed. I'm just not sure where we could actually put the throttle to achieve that.

File: src/components/InputOption.js [hunk 1]
```diff
@@ -6,6 +6,25 @@ import {
...
+
+    if (optionType === 'number') {
+      value = value === '' ? '' : parseFloat(value)
+    }
+
+    attrs[key] = value
+    if (option.onChange !== undefined) {
+      attrs = option.onChange(attrs, model)
+    }
+    this.props.onChange(attrs)
```
Comment (jeffeb3) [inline src/components/InputOption.js:25]: I think we could probably do this without a doubled state by setting a flag on attrs.

File: src/components/InputOption.js [hunk 2]
```diff
@@ -4,8 +4,18 @@ import {
   Form,
   Row
 } from 'react-bootstrap'
+import debounce from 'lodash/debounce'
 
 class InputOption extends Component {
+  constructor(props) {
+    super(props);
+    this.delayedSet = debounce( (value, key, onChange) => {
```
Comment (jeffeb3) [inline src/components/InputOption.js:12]: This is a much cleaner option. Thanks Bob.

File: src/components/InputOption.js [hunk 3]
```diff
@@ -4,8 +4,18 @@ import {
...
+      let attrs = {}
+      attrs[key] = value
+      onChange(attrs)
+    }, 1500)
```
Comment (jeffeb3) [inline src/components/InputOption.js:16]: WDYT about this timeout? It is a bit long, but I liked it not computing while I was typing.

Comment (bobnik) [reply src/components/InputOption.js:16]: 👍🏻

**Pull request merged**